dot/parachain/util/util.go

```diff
@@ -25,3 +47,99 @@ func SigningKeyAndIndex(
 	}
 	return nil, 0
 }
+
+// DetermineNewBlocks determines the hashes of all new blocks we should track metadata for, given this head.
+//
+// This is guaranteed to be a subset of the (inclusive) ancestry of `head` determined as all
+// blocks above the lower bound or above the highest known block, whichever is higher.
+// This is formatted in descending order by block height.
+//
+// An implication of this is that if `head` itself is known or not above the lower bound,
+// then the returned list will be empty.
+//
+// This may be somewhat expensive when first recovering from major sync.
+//
+// NOTE: TOTO: this issue needs to be finished, see issue #3933
```

Comment:
```suggestion
// NOTE: this issue needs to be finished, see issue #3933
```

Reply:
what else needs to be finished though? Can clarify in the issue.

Reply:
Added comment to issue.